### PR TITLE
🐛 Source Sftp-Bulk: fix empty password field exception

### PR DESCRIPTION
**Pull request by SoerenStahlmann**
## What
When creating a SFTP-Bulk connection you can choose to authenticate via password or a ssh-key. If you choosing to authenticate with a ssh-key an exception is thrown when the password field is empty, since the dictionary value 'password' does not exist in that case. 

This case was already handled for the other authentication method by providing a default 'None' value when getting the dictionary key for an empty ssh-key.

## How
I provided a default value in case the password field was not set

```python
password=config.get("password", None),
```

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

### CODE REVIEW

File: docs/integrations/sources/sftp-bulk.md [hunk 1]
```diff
@@ -120,6 +120,7 @@ More formats \(e.g. Apache Avro\) will be supported in the future.
 
 | Version | Date       | Pull Request                                              | Subject                       |
 | :------ | :--------- | :-------------------------------------------------------- | :---------------------------- |
+| 0.1.3   | 2023-11-22 |  | Password field no longer throws exception when empty |
```
Comment (marcosmarxm) [inline docs/integrations/sources/sftp-bulk.md:123]: Please update here.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Small change.

**Review by marcosmarxm — COMMENTED** [review]
At the moment integration tests are failing and will take some time to fix them. Thanks for your patience.

**Comment by SoerenStahlmann** [comment]
Are there any news concerning the issue @marcosmarxm? We are currently running a forked version of the connector and would rather switch to the official connector as soon as the issue is merged.

**Comment by marcosmarxm** [comment]
@SoerenStahlmann sorry the lack of updates here. The latest version of this connector already fix this issue. Please upgrade to the latest version.

**Pull request closed without merging**